### ISSUE TITLE
batch size for asr model 

### ISSUE THREAD

**Issue by zhan0502** [NVIDIA/NeMo#2902]
I noticed that for ASR training batch size setup, I noticed that there is a trick to set the accumulate_grad_batches  value to cater for large batches, the performance is still quite different between set batch size = 32 vs. set batch size =16 with   accumulate_grad_batches = 2. Although in theory it should work, but when I do the training for  QuartzNet15x5-Zh-NR.nemo,  I found if using batch size = 32, I can get loss < 20 in about 10 epochs, whereas if I use  batch size =16 with   accumulate_grad_batches = 2, I train for 100 epochs, the loss level is at 67 (which means the training diverges). 

Does this mean, we would have to use high RAM GPUs to train the models? Does anyone observe the similar issue before?

below is a sample setup for batch size, this is because my server couldn't cater for batchsize =32, therefore, I use this trick, but the training metrics was very bad. 
 cfg = copy.deepcopy(char_model.cfg)
 with open_dict(cfg):    
   cfg.train_ds.manifest_filepath = f"{train_manifest_cleaned},{dev_manifest_cleaned}"
  cfg.train_ds.labels = list(train_dev_set)
  cfg.train_ds.normalize_transcripts = False
  cfg.train_ds.batch_size = 16
  cfg.train_ds.num_workers = 8
  cfg.train_ds.pin_memory = True
  cfg.train_ds.trim_silence = True 
trainer = ptl.Trainer(gpus=4, 
                      accelerator='ddp',
                      max_epochs=EPOCHS, 
                      accumulate_grad_batches=2,
                      checkpoint_callback=False,
                      logger=False,
                      log_every_n_steps=5,
                      check_val_every_n_epoch=10)

**Comment by titu1994** [comment]
I would suggest trying branch 16 without any accumulate grad to see if it converges or not. We don't have many experiments with accumulate grad > 1, but the few that we do generally worked close to similar results (however we use it on already large global batch sizes of > 1k). I would also be concerned by LR for finetuning of QuartzNet, but if BS 32 converges stably then it is not a major factor contributing to accum grad issue

**Comment by zhan0502** [comment]
Thanks, I'll retrain with bs=16 and accumulate_grad_batches=1 to see if there is any difference.

**Comment by zhan0502** [comment]
Hi, I have test out if use bs=16  and accumulate_grad_batches=1 only, the model training will not converage, after 300 epochs (learning rate =0.005) loss is still at 40+.  The server I'm using for this training is T4 and there is no other application running in this server.  Would appreciate your comments on the following:
1) besides setting the parameter accumulate_grad_batches, is there any  other place I could config, if we still want to make the batch size =32? In pytorch lightening, accumulate_grad_batches is used to bring up the effective batch size. https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html. But in this case, the effective batch size seems not quite the same as the actual batch size.. 
2) if 1) is not possible, is any other tricks we could use for the bs=16 for quartznet training? For example, would reduce the learn rate to very low possibly work?

**Comment by titu1994** [comment]
Loss at 40 ish sounds reasonable to me, since it uses spec cutout. If the word error rate is not sufficient then perhaps using another model would be a better choice. 

If quartNet must be used, you could try freezing the encoder, unfreezing the batch norm layers and finetuning just the decoder layer. 

If even that doesn't work, you could try lower LR than 0.005 but I would not expected good results
